scripts/update_models.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+set -e
+
+usage()
+{
+  echo "Usage: update_models.sh -c [ -v MODELS_VERSION_NUMBER ]"
+  exit 2
+}
+
+get_aws_sdk_go()
+{
+  # clone aws-sdk-go into folder
+  git clone https://github.com/aws/aws-sdk-go.git $1
+  if [ -n "$2" ]; then
+      pushd $1
+      git checkout $2
+      popd
+  fi
+  return 0
+}
+
+copy_model_files()
+{
+    rm -rf $2/apis/*
+    rm -rf $2/endpoints/*
+    cp -R $1/apis/* $2/apis/
+    cp -R $1/endpoints/* $2/endpoints/
+    rm $2/apis/*.go
+    rm $2/endpoints/*.go
+    return 0
+}
+
+build_files()
+{
+    # build the code generator and run it
+    echo "Build the code generator"
+    swift build --product aws-sdk-swift-codegen
+    echo "Run the code generator"
+    swift run
+    echo "Compile service files"
+    # build services after having generated the files
+    swift build
+}
+
+check_for_local_changes()
+{
+    LOCAL_CHANGES=`git status --porcelain`
+    if [ -n "$LOCAL_CHANGES" ] && [ "$COMMIT_CHANGES" == 1 ]; then
+        echo "You have local changes already and you have requested to commit changes after this script has run.\nEither remove the -c option or stash your local changes."
+        usage
+        exit
+    fi
+}
+
+commit_changes()
+{
+    COMMIT_MSG="Sync models with aws-sdk-go "$AWS_MODELS_VERSION
+    git add models
+    git add Sources/AWSSDKSwift
+    git commit -m "$COMMIT_MSG"
+}
+
+cleanup()
+{
+    if [ -n "$TEMP_DIR" ]; then
+        rm -rf $TEMP_DIR
+    fi
+}
+
+while getopts 'cv:' option
+do
+    case $option in
+        v) AWS_MODELS_VERSION=$OPTARG ;;
+        c) COMMIT_CHANGES=1 ;;
+        *)
+        usage
+        exit ;;
+    esac
+done
+
+
+trap cleanup EXIT
+
+check_for_local_changes
+
+#create temp folder
```

Comment:
I'd suggest removing these, your echo statements do a great job of describing what's going on already.